src/common/gitApiExtensions.ts

```diff
@@ -1,19 +1,25 @@
 import { Uri } from 'vscode';
 import * as cp from 'child_process';
 
-interface BaseBaseRepository {
+// This refers to Repository in
+// vscode/extension/git/src/git.ts
+interface GitRepository {
```

Comment:
nit: keep 'BaseBase'
It refers to the nested nature of the API-hack. Yes, it's ugly, but it should still be informative

Reply:
I changed it because when I was following this file. I found it extremely difficult to understand the meaning of BaseBase. I think it's easier to understand with the categorical name like this interface is about the git operations and also in their `git.ts`
But I am also okay with reverting the name since adding those comments should help finding the file the interface is referring to.

Reply:
All the Repository types 'are about' git operations, and BaseRepository is the name used in 'extension/git' at all levels `api -> repository -> git`
So calling one of them GitRepository and not changing the others doesn't work.
`Repository -> BaseRepository -> BaseBaseRepository`

Another consistent naming scheme could maybe be: `(Git)ApiRepository, Repository, and GitRepository`
But that means chaning the api type name, which is not a good idea.

The comments are helpful 👍

Reply:
Ah..I understand where you coming from now. I agreed that it would have been so much easier if we could change the api type name, but that's another can of worms.

I guess I will take one last stab at this, and naming is hard. How about this?
Original: `Repository -> BaseRepository -> BaseBaseRepository`
New: `Repository -> BaseRepository -> BaseGitRepository`

If that's still not resonating, I can revert the name back. I am glad at least there's this conversation so people can refer back to.

Reply:
BaseGitRepository sounds good 👍 
>  I am glad at least there's this conversation so people can refer back to.

yes :)